.github/workflows/e2e-main.yml

```diff
@@ -0,0 +1,161 @@
+      ORGANIZATION: ${{ inputs.organization || 'crc-org' }}
+      BRANCH: ${{ inputs.branch || 'main' }}
+    steps:
+      - uses: actions/checkout@v4
+        if: github.event_name == 'workflow_dispatch' || github.event_name == 'workflow_run'
+        with:
+          repository: ${{ env.ORGANIZATION }}/${{ env.REPO }}
+          ref: ${{ env.BRANCH }}
+          path: ${{ env.REPO }}
+
+      - uses: actions/checkout@v4
+        if: github.event_name == 'push'
+        with: 
+          path: crc-extension
+
+      - name: Setup default repository name
+        env: 
+          EVENT: ${{ github.event_name }}
+        run: |
+          repository=crc-extension
+          if [[ "$EVENT" == 'workflow_dispatch' || "$EVENT" == 'workflow_run' ]]; then
+            repository=${{ env.REPO }}
+          fi
+          echo "REPOSITORY=$repository" >> $GITHUB_ENV
+      # Checkout podman desktop
+      - uses: actions/checkout@v4
+        with:
+          repository: containers/podman-desktop
+          ref: main
+          path: podman-desktop
+
+      - uses: pnpm/action-setup@v4
+        name: Install pnpm
+        with:
+          run_install: false
+          package_json_file: ./${{ env.REPO }}/package.json
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Update podman v. 5.x
+        run: |
+          # ubuntu version from kubic repository to install podman we need (v5)
+          ubuntu_version='23.04'
+          sudo sh -c "echo 'deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_${ubuntu_version}/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:unstable.list"
+          curl -L "https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_${ubuntu_version}/Release.key" | sudo apt-key add -
+          # install necessary dependencies for criu package which is not part of 23.04
+          sudo apt-get install -qq libprotobuf32t64 python3-protobuf libnet1
+          # install criu manually from static location
+          curl -sLO http://cz.archive.ubuntu.com/ubuntu/pool/universe/c/criu/criu_3.16.1-2_amd64.deb && sudo dpkg -i criu_3.16.1-2_amd64.deb
+          sudo apt-get update -qq
+          sudo apt-get -qq -y install podman || { echo "Start fallback steps for podman nightly installation from a static mirror" && \
+            sudo sh -c "echo 'deb http://ftp.lysator.liu.se/pub/opensuse/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_${ubuntu_version}/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:unstable.list" && \
+            curl -L "http://ftp.lysator.liu.se/pub/opensuse/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_${ubuntu_version}/Release.key" | sudo apt-key add - && \
+            sudo apt-get update && \
+            sudo apt-get -y install podman; }
+          podman version
+      - name: Revert unprivileged user namespace restrictions in Ubuntu 24.04
```

Comment:
this step is part of podman 5 installation, if podman is not needed, we can also remove this one.